### DISCUSSION TITLE
Blocking specific days of the week

### DISCUSSION THREAD

**Discussion by t-barclay** [easepick/easepick#5] [Q&A] [answered]
Is it possible to lock multiple date ranges using the LockPlugin? As a hotel calendar picker, in addition locking dates in the past we would also like to lock dates in the future that we are closed.

**Comment by wakirin** [comment, marked as answer]
Like this https://jsfiddle.net/waki/jftLm3du/ ?

**Reply by wakirin** [reply]
Perhaps you want to disable Saturday, Sunday. 
Then you can do like this: https://jsfiddle.net/waki/zoL3dgbp/

**Reply by t-barclay** [reply]
That is definitely helpful, although is it possible to do exactly that but to grey those dates out rather than display the orange hash? A distinction between dates that are booked vs closed would be helpful for users

**Reply by wakirin** [reply]
You can change classes on `view` event.

https://jsfiddle.net/waki/n3y4mwtq/

**Reply by t-barclay** [reply]
That's perfect. Thank you very much!

**Reply by t-barclay** [reply]
I have two more twists in the plot that I'd be super grateful for some help with.... whilst the hotel is closed Mon, Tues and Weds nights which I have blocked out as above, if a user is trying to book Sunday night they are unable to select Monday as the end date. Is there a way around this?

Lastly, there is a two night minimum booking for Saturday and Sunday night only. Is it possible to set `minDays` only for those days of the week?

**Reply by wakirin** [reply]
> Lastly, there is a two night minimum booking for Saturday and Sunday night only. Is it possible to set minDays only for those days of the week?

I assume that this current version may require a lot of additional code to implement it.
I think I can make a few changes to make this implementation easier.
Currently I'm working on an update (1.0.2).
It's in beta now.
I plan to release (1.0.2) next week.

**Reply by wakirin** [reply]
@t-barclay 
1.0.2 is now available. You can do something like this: https://jsfiddle.net/waki/4vmo7bew/
Also I would like to provide another solution where the user doesn't have to do an extra click on Sunday: https://jsfiddle.net/waki/y6kdotj8/

**Reply by t-barclay** [reply]
@wakirin thank you very much for the update. Neither of the two jsfiddles seem to be restricting weekend bookings to 2 nights, and Monday isn't available as a checkout day but not a checkin day. Please forgive me if I've misunderstood something?

**Reply by wakirin** [reply]
Perhaps I misunderstood the case. Could you explain what you expected to see?

**Reply by t-barclay** [reply]
There are two points:

1. We are closed for check-ins on Monday, Tuesday and Wednesday. However, a guest can still check out on a Monday. In the demo, Monday is locked so that the user cannot choose this as a start date (check-in), but it should be available as an end date (check-out).
2. We have no minimum night stay, apart from Saturday and Sunday nights which must be booked together as a two night stay. Below are a few example situations:

- if the user selects Thursday or Friday as their start date, only Friday (if start date is Thursday) and Saturday should be available as end dates.
- if the user chooses Saturday as the start date, only Monday should be available as the end date.
- Sunday should not be available as a start date

**Reply by wakirin** [reply]
Perhaps this will help: https://jsfiddle.net/waki/3mt0zfdn/

**Comment by crossiono** [comment]
good afternoon,  @wakirin 
my situation is they can only book saturday to saturday.  i have the calendar to only be able to select saturdays, but i want the bookedDates to retain the locked style.
